rfd/xxxx-azure-node-join.md

```diff
@@ -0,0 +1,130 @@
+Product: @klizhentas && @xinding33
+Security: @reedloden
+
+## What
+
+Teleport nodes running on Azure virtual machines will be able to join a
+Teleport cluster without the need to share any secret token with the node.
+Instead, the node will present a signed
+[attested data document](https://learn.microsoft.com/en-us/azure/virtual-machines/linux/instance-metadata-service?tabs=linux#attested-data)
+to confirm the subscription the VM is running in. With a configured Azure join
+token on the auth server, the node will be allowed to join the cluster based
+on its Azure subscription.
+
+This is the Azure equivalent of EC2/IAM join as described in
+[RFD 41](https://github.com/gravitational/teleport/blob/master/rfd/0041-aws-node-join.md).
+
+## Why
+
+This will make provisioning nodes on Azure simpler and arguably more secure.
+There is no need to configure static tokens, or to manage the provisioning and
+deployment of dynamic tokens.
+
+## Details
+
+In place of a join token, nodes will present a signed attested data document
+to the auth server. The document will be fetched on the nodes via instance
+metadata. The request can include a user-specified nonce which will also be
+signed. Like the custom headers in IAM join, the nonce allows Teleport to issue
+a challenge (crypto random string of bytes) that the node must include in the
+signed request. We can use gRPC bidirectional streams to implement the
+following flow at a new `RegisterWithAzureToken` gRPC endpoint:
+
+1. The Node initiates a `RegisterWithAzureToken` request.
+2. The auth server enforces that TLS is used for transport.
+3. The auth server sends a base64 encoded 24 byte crypto-random challenge.
+  - We use 24 bytes instead of 32 because the nonce parameter is limited to 32
+    characters, and 24 bytes require 32 base64 characters.
+4. The node requests the attested data document using the challenge as the
+  nonce. Azure returns the signed document.
+5. The node sends a join request to the auth server, including the signed
+  document and other paramaters in a `RegisterUsingTokenRequest`.
+  - `RegisterUsingTokenRequest` will need to be extended to include info not in
+  the attested data document (notably the VM's resource group and region).
+6. The auth server checks that signed document is valid:
+  - The document's pkcs7 signature is valid.
+  - The returned nonce matches the issued challenge.
+  - The requested token allows the node to join.
+7. The auth server sends credentials to join the cluster.
+
+As with IAM join, the flow will occur within a single streaming gRPC request and
+only 1 attempt with a given challenge will be allowed, with a 1 minute timeout.
+
+### Teleport Configuration
+
+The existing provision token type can be extended to support Azure
+authentication, using new Azure-specific fields in the token rules section.
+
+```yaml
+kind: token
+version: v2
+metadata:
+  name: example_azure_token
+spec:
+  roles: [Node,Kube,Db]
+  allow:
```

Comment:
the other new join methods are putting their config inside a specific block for that method, rather than the top level allow block. check the k8s RFD for an example https://github.com/gravitational/teleport/blob/master/rfd/0094-kubernetes-node-joining.md#teleport-join-tokens